### ISSUE TITLE
Error while executing policy flow-logs-enabled

### ISSUE THREAD

**Issue by selvaranjith92** [cloud-custodian/cloud-custodian#6152]
```custodian run --dry-run -s /tmp/. --region us-east-1 vpc-flow-log.yaml 
2020-09-24 13:59:57,695: custodian.output:ERROR Error while executing policy
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/c7n/policy.py", line 281, in run
    resources = self.policy.resource_manager.resources()
  File "/usr/local/lib/python3.7/dist-packages/c7n/query.py", line 499, in resources
    resources = self.filter_resources(resources)
  File "/usr/local/lib/python3.7/dist-packages/c7n/manager.py", line 111, in filter_resources
    resources = f.process(resources, event)
  File "/usr/local/lib/python3.7/dist-packages/c7n/filters/core.py", line 359, in process
    return self.process_set(resources, event)
  File "/usr/local/lib/python3.7/dist-packages/c7n/filters/core.py", line 378, in process_set
    resources = f.process(resources, event)
  File "/usr/local/lib/python3.7/dist-packages/c7n/resources/vpc.py", line 139, in process
    fl['LogDestination'], destination)
KeyError: 'LogDestination'
2020-09-24 13:59:57,697: custodian.commands:ERROR Error while executing policy flow-logs-enabled, continuing
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/c7n/commands.py", line 272, in run
    policy()
  File "/usr/local/lib/python3.7/dist-packages/c7n/policy.py", line 1169, in __call__
    resources = PullMode(self).run()
  File "/usr/local/lib/python3.7/dist-packages/c7n/policy.py", line 281, in run
    resources = self.policy.resource_manager.resources()
  File "/usr/local/lib/python3.7/dist-packages/c7n/query.py", line 499, in resources
    resources = self.filter_resources(resources)
  File "/usr/local/lib/python3.7/dist-packages/c7n/manager.py", line 111, in filter_resources
    resources = f.process(resources, event)
  File "/usr/local/lib/python3.7/dist-packages/c7n/filters/core.py", line 359, in process
    return self.process_set(resources, event)
  File "/usr/local/lib/python3.7/dist-packages/c7n/filters/core.py", line 378, in process_set
    resources = f.process(resources, event)
  File "/usr/local/lib/python3.7/dist-packages/c7n/resources/vpc.py", line 139, in process
    fl['LogDestination'], destination)
KeyError: 'LogDestination'
2020-09-24 13:59:57,698: custodian.commands:ERROR The following policies had errors while executing
 - flow-logs-enabled```


And my code is 


```policies:
  - name: flow-logs-enabled
    resource: vpc
    filters:
      - not:  
        - type: flow-logs
          destination: "vpc-flow-logs"
          enabled: true
          set-op: or
          op: equal
          traffic-type: all
          status: active
          destination-type: s3
      - IsDefault: true```


@kapilt please help me on this. I need to filter the flow logs with the destination (LogDestination) name.

**Comment by selvaranjith92** [comment]
@kapilt please help me on this. I need to filter the flow logs with the destination (LogDestination) name.

**Comment by PratMis** [comment]
What's your end goal here? If it is to filter flow logs with destination type i.e. s3, you can use the below shown policy:
```
- name: vpc-flow-logs-s3
  resource: vpc
  filters:
    - or:
      - type: flow-logs
        enabled: false
      - not:
        - type: flow-logs
          enabled: true
          destination-type: s3
```
I can you can add one more filter for cw destination here

**Comment by selvaranjith92** [comment]
@PratMis  As we are using two different  LogDestination in s3. So i want to include the "destination" field as well in my custodian policy.

**Comment by kapilt** [comment]
@selvaranjith92 what version of custodian is this please include output of `custodian version --debug`

**Comment by selvaranjith92** [comment]
@kapilt PFB the details,

```
custodian version --debug

Please copy/paste the following info along with any bug reports:

Custodian:   0.9.6
Python:      3.7.3 (default, Jul 25 2020, 13:03:44) 
             [GCC 8.3.0]
Platform:    posix.uname_result(sysname='Linux', nodename='myapp', release='4.14.173-137.229.amzn2.x86_64', version='#1 SMP Wed Apr 1 18:06:08 UTC 2020', machine='x86_64')
Using venv:  False
Docker: True
Installed: 

argcomplete==1.12.0
attrs==20.2.0
boto3==1.15.4
botocore==1.18.4
importlib-metadata==2.0.0
jmespath==0.10.0
jsonschema==3.2.0
pyrsistent==0.17.3
python-dateutil==2.8.1
pyyaml==5.3.1
s3transfer==0.3.3
setuptools==40.8.0
six==1.12.0
tabulate==0.8.7
urllib3==1.25.10
zipp==3.2.0
```

**Comment by selvaranjith92** [comment]
@kapilt Please help on the above issue. I need to filter the configuration with `destination` in vpc flow logs.

**Comment by kapilt** [comment]
fix merged, per #7569
